4_naming_conventions/43_file_names.md

```diff
@@ -0,0 +1,211 @@
+digits followed by a period followed by one or more letters or digits.
+
+### 4.3.5 File naming guidelines for modules
+
+Below sections are the file naming guidelines for EDK II meta files and source files. The
+guidelines are not just considering the the uniformity of file naming, but it also provides
+the flexibility of file name construction for the scenario of different EDK II module
+designs; such as the support for multiple processor architectures and vendors. It may require the
+further discussions between EDK II maintainers and contributors in order to determine the best
+naming of the EDK II module file.
+
+#### 4.3.5.1 EDK II meta files within a package
+
+```
+<PackageName>Pkg.dec
+<PackageName>Pkg.dsc
+
+   <PackageName> REQUIRED  *
+```
+
+#### 4.3.5.2 EDK II INF file within a Module instance
+* If the implementation is general for all CPU architectures without the specific arch dependency:
+```
+<Feature><Phase>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm,
+                              Smm, Uefi.
+
+Example:
+   - SmbiosDxe.inf
+```
+
+* If the implementation of CPU arch or vendor shares some code in the module. Or the implementation is only for the specific CPU arch or vendor:
+```
+<Feature><Phase><CpuArch><Vendor>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                              StandaloneMm, Smm, Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+
+Example:
+   - CpuIo2DxeAmd.inf
+   - CpuIo2DxeIntel.inf
+   - CpuIo2DxeIa32_X64.inf
+
+```
+
+* If the implementation does not have any shared code between phases:
+
+```
+<Feature>/<Phase>/<Feature>.inf
+  or
+<Feature>/<Phase>/<Feature><CpuArch><Vendor>.inf
+
+   <Feature>      OPTIONAL    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm, Smm,
+                              Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'   
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+Example:
+   PCD/Dxe/Pcd.inf
+```
+
+#### 4.4.5.3 EDK II INF file within a Library instance
+```
+<Phase><CpuArch><Vendor><LibraryClassName><Dependency>.inf

```

Comment:
@niruiyu
In order to have a consistent naming rule for \<CpuArch\>\<Vendor\>, I will apply [\<CpuArch\>[\<Vendor\>]] on this case as well. A long file name for \<CpuArch\>\<Vendor\> is expected.

Reply:
> @niruiyu
> In order to have a consistent naming rule for <CpuArch><Vendor>, I will apply [<CpuArch>[<Vendor>]] on this case as well. A long file name for <CpuArch><Vendor> is expected.

@niruiyu I still use [\<CpuArch\>][\<Vendor\>], so Vendor is unnecessary to have a dependence on CpuArch. The reason is the module is expected for X86 only, that is no need to have Ia32X64 for the vendor INF under X86 module. This concept is applied to the entire File name section. Please check the next version and see what else we can improve.

Reply:
> Please check the next version and see what else we can improve.

PR is updated.